workflow-controller/controller-service/src/main/java/io/hyscale/controller/util/WorkflowUtil.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.util;
+
+import java.io.File;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.controller.core.exception.ControllerErrorCodes;
+import io.hyscale.controller.model.WorkflowContext;
+import io.hyscale.servicespec.commons.exception.ServiceSpecErrorCodes;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+
+@Component
+public class WorkflowUtil {
+    
+    @Autowired
+    private ServiceSpecMapper serviceSpecMapper;
+
+    public WorkflowContext getWorkflowContext(String appName, String serviceSpecPath,
+            Map<String, String> serviceProfileMap) throws HyscaleException {
+        if (StringUtils.isBlank(serviceSpecPath)) {
+            throw new HyscaleException(ServiceSpecErrorCodes.SERVICE_SPEC_REQUIRED);
+        }
+        if (StringUtils.isBlank(appName)) {
+            throw new HyscaleException(ControllerErrorCodes.APP_NAME_REQUIRED);
+        }
+        String serviceName = ServiceSpecUtil.getServiceNameFromPath(serviceSpecPath);
+        String profilePath = serviceProfileMap != null ? serviceProfileMap.remove(serviceName) : null;
+        File serviceSpecFile = new File(serviceSpecPath);
+        File profileFile = null;
+        if (profilePath != null) {
+            profileFile = new File(profilePath);
+        }
+        WorkflowContext workflowContext = new WorkflowContext();
+        workflowContext.setServiceName(serviceName);
+        ServiceSpec serviceSpec = serviceSpecMapper.from(serviceSpecFile, profileFile);
+        workflowContext.setServiceSpec(serviceSpec);
+        workflowContext.setAppName(appName.trim());
+        workflowContext.setEnvName(CommandUtil.getEnvName(profilePath, appName.trim()));
+        return workflowContext;
```

Comment:
Let the workflow context building be in the invoker , or else the context building is broken down into chunks in diff invokers. Either build all (min required fields) at one place or be clear it build at the invoker itself. Also this should just process the service spec and return.